src/urllib3/connection.py

```diff
@@ -548,25 +595,24 @@ def set_cert(
         self.ca_cert_data = ca_cert_data
 
     def connect(self) -> None:
-        self._connecting_to_proxy = bool(self.proxy)
-
         sock: Union[socket.socket, "ssl.SSLSocket"]
         self.sock = sock = self._new_conn()
         server_hostname: str = self.host
         tls_in_tls = False
 
-        if self._is_using_tunnel():
-            if self.tls_in_tls_required:
+        # Do we need to establish a tunnel?
+        if self._tunnel_host is not None:
+            # We're tunneling to an HTTPS origin so need to do TLS-in-TLS.
+            if self._tunnel_scheme == "https":
                 self.sock = sock = self._connect_tls_proxy(self.host, sock)
                 tls_in_tls = True
 
-            self._connecting_to_proxy = False
+            # If we're tunneling it means we're connected to our proxy.
+            self._has_connected_to_proxy = True
 
-            self._tunnel()
+            self._tunnel()  # type: ignore[attr-defined]
```

Comment:
Why is mypy complaining here?

Reply:
`_tunnel` isn't a part of the [typeshed definition for `http.client.HTTPConnection`](https://github.com/python/typeshed/blob/805519717b7213ed232593931df815fd6c839ff5/stdlib/http/client.pyi#L139).